### PR TITLE
Initialize committee pubkey with genesis shard state; Change bls address to pubkey in staking contract.

### PR DESCRIPTION
**Pull request by rlan35**
Our previous setup is to populate consensus committee keys on the fly based on ping pong message, which was a temporary solution to make testnet work. This PR added the necessary code to setup the initial committee based on genesis shard state. This PR should be backward compatible as the ping/pong logic are not completely removed yet.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -256,8 +258,6 @@ func New(host p2p.Host, consensusObj *consensus.Consensus, db ethdb.Database, is
 			os.Exit(1)
 		}
 		node.blockchain = chain
-		// Store the genesis shard state into db.
-		node.blockchain.StoreNewShardState(node.blockchain.CurrentBlock(), nil)
```
Comment (LeoHChen) [inline node/node.go]: why this is removed?

File: node/staking.go [hunk 2]
```diff
@@ -44,8 +46,12 @@ func (node *Node) UpdateStakingList(stakeInfoReturnValue *structs.StakeInfoRetur
 		}
 		// True if the token is still staked within the locking period.
 		if curEpoch-startEpoch <= lockPeriodCount.Uint64()*lockPeriodInEpochs {
+			blsPubKey := types.BlsPublicKey{}
+			copy(blsPubKey[:32], stakeInfoReturnValue.BlsPubicKeys1[i][:])
```
Comment (LeoHChen) [inline node/staking.go]: why copy 3 part?

File: consensus/consensus.go [hunk 3]
```diff
@@ -499,6 +513,12 @@ func (consensus *Consensus) DebugPrintValidators() {
 func (consensus *Consensus) UpdatePublicKeys(pubKeys []*bls.PublicKey) int {
 	consensus.pubKeyLock.Lock()
 	consensus.PublicKeys = append(pubKeys[:0:0], pubKeys...)
+	consensus.CommitteeAddresses = map[common.Address]bool{}
+	for _, pubKey := range consensus.PublicKeys {
+		consensus.CommitteeAddresses[utils.GetBlsAddress(pubKey)] = true
```
Comment (chaosma) [inline consensus/consensus.go]: if a validator not in the committee, does its key not exist here or value is false?

File: core/types/shard_state.go [hunk 4]
```diff
@@ -11,10 +12,13 @@ import (
 // ShardState is the collection of all committees
 type ShardState []Committee
 
+// BlsPublicKey defines the bls public key
+type BlsPublicKey [96]byte
+
 // NodeID represents node id (BLS address).
 type NodeID struct {
 	EcdsaAddress string
```
Comment (chaosma) [inline core/types/shard_state.go]: may use Address type instead of string as well

File: node/node.go (hunk 1 above)
Comment (rlan35) [reply node/node.go]: It's not removed. it's moved to the other method InitGenesisShardState

File: node/staking.go (hunk 2 above)
Comment (rlan35) [reply node/staking.go]: because public key is 96 bytes and solidity doesn't support long byte array very well

File: consensus/consensus.go (hunk 3 above)
Comment (rlan35) [reply consensus/consensus.go]: Just not exist, there won't be false value. The map is used as a set

File: core/types/shard_state.go (hunk 4 above)
Comment (rlan35) [reply core/types/shard_state.go]: using fixed type is better than using string. Easier to debug.

**Comment by rlan35** [comment]
> Should add the flag to disable the original ping/pong message, to make sure the new mechanism works. How do you solve the problem of the new node then?

New node should still work. As I didn't change the previous pingpong logic.

**Pull request merged**